homeassistant/components/fritz/switch.py

```diff
@@ -0,0 +1,644 @@
+    fritzbox_tools: FritzBoxTools,
+    service_name: str,
+    service_suffix: str | None,
+    action_name: str,
+    **kwargs: Any,
+) -> None | dict:
+    """Return service details."""
+    return await fritzbox_tools.hass.async_add_executor_job(
+        partial(
+            service_call_action,
+            fritzbox_tools,
+            service_name,
+            service_suffix,
+            action_name,
+            **kwargs,
+        )
+    )
+
+
+def service_call_action(
+    fritzbox_tools: FritzBoxTools,
+    service_name: str,
+    service_suffix: str | None,
+    action_name: str,
+    **kwargs: Any,
+) -> dict | None:
+    """Return service details."""
+
+    if f"{service_name}{service_suffix}" not in fritzbox_tools.connection.services:
+        return None
+
+    try:
+        return fritzbox_tools.connection.call_action(
+            f"{service_name}:{service_suffix}",
+            action_name,
+            **kwargs,
+        )
+    except FritzSecurityError:
+        _LOGGER.error(
+            "Authorization Error: Please check the provided credentials and verify that you can log into the web interface",
+            exc_info=True,
+        )
+        return None
+    except (FritzActionError, FritzActionFailedError, FritzServiceError):
+        _LOGGER.error(
+            "Service/Action Error: cannot execute service %s",
+            service_name,
+            exc_info=True,
+        )
+        return None
+    except FritzConnectionException:
+        _LOGGER.error(
+            "Connection Error: Please check the device is properly configured for remote login",
+            exc_info=True,
+        )
+        return None
+
+
+def get_deflections(
+    fritzbox_tools: FritzBoxTools, service_name: str
+) -> list[OrderedDict[Any, Any]] | None:
+    """Get deflection switch info."""
+
+    deflection_list = service_call_action(
+        fritzbox_tools,
+        service_name,
+        "1",
+        "GetDeflections",
+    )
+
+    if not deflection_list:
+        return []
+
+    deflections: list = []
+    deflections.append(
+        xmltodict.parse(deflection_list["NewDeflectionList"])["List"]["Item"]
+    )
+
+    return deflections
+
+
+def deflection_entities_list(
+    fritzbox_tools: FritzBoxTools, device_friendly_name: str
+) -> list[FritzBoxDeflectionSwitch]:
+    """Get list of deflection entities."""
+
+    _LOGGER.debug("Setting up %s switches", SWITCH_TYPE_DEFLECTION)
+
+    entities_list: list = []
+    service_name = "X_AVM-DE_OnTel"
+    deflections_response = service_call_action(
+        fritzbox_tools, service_name, "1", "GetNumberOfDeflections"
+    )
+    if not deflections_response:
+        _LOGGER.debug("The FRITZ!Box has no %s options", SWITCH_TYPE_DEFLECTION)
+        return []
+
+    _LOGGER.debug(
+        "Specific %s response: GetNumberOfDeflections=%s",
+        SWITCH_TYPE_DEFLECTION,
+        deflections_response,
+    )
+
+    if deflections_response["NewNumberOfDeflections"] != 0:
```

Comment:
```suggestion
    if deflections_response["NewNumberOfDeflections"] == 0:
        return entities_list
```
Could return early and outdent the below

Reply:
Done